### PR TITLE
Python infra: run CCF nodes inside containers

### PR DESCRIPTION
**Pull request by jumaffre**
Part of #2608 (see issue description for rationale behind this).

TODO: 
- [x] Fix network for parallel tests
- [x] Fix `lts_compatibility` test 
- [ ] Fix access to JWT issuer service (to be addressed in a separate PR)

### CODE REVIEW

File: tests/infra/remote_shim.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) Microsoft Corporation. All rights reserved.
+`# Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (achamayou) [inline tests/infra/remote_shim.py:1]: o_O`

File: tests/infra/remote_shim.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+
+class PassThroughShim(infra.remote.CCFRemote):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+
+# Current limitations, which should be overcomable:
+# 1. No support for SGX kernel built-in support (i.e. 5.11+ kernel) in Docker environment
+# 2. No support for concurrent test: all nodes containers attach to the same network at pre-defined IPs
```
Comment (jumaffre) [inline tests/infra/remote_shim.py:57]: Current limitations are listed here, but I'd like to address them in a separate PR.

File: tests/e2e_logging.py [hunk 3]
```diff
@@ -198,21 +175,14 @@ def test_clear(network, args):
...
                         check(c.get(f"{resource}?id={log_id}"), result={"msg": msg})
-                    check(
-                        c.delete(f"{resource}/all"),
-                        result=None,
-                    )
+                    check(c.delete(f"{resource}/all"), result=None)
                     for log_id in log_ids:
                         get_r = c.get(f"{resource}?id={log_id}")
                         if args.package == "libjs_generic":
-                            check(
```
Comment (achamayou) [inline tests/e2e_logging.py:212]: Are these formatting changes necessary?

File: CMakeLists.txt [hunk 4]
```diff
@@ -623,6 +623,7 @@ if(BUILD_TESTS)
     NAME e2e_logging_cft
     PYTHON_SCRIPT ${CMAKE_SOURCE_DIR}/tests/e2e_logging.py
     CONSENSUS cft
+    DOCKER_NODES TRUE
```
Comment (achamayou) [inline CMakeLists.txt:626]: Can we run this in Daily only? I imagine it's going to be slower, and possibly less stable than the current setup at least at first.

**Review by achamayou — APPROVED** [review]

**Pull request merged**